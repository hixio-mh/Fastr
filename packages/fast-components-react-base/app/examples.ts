import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export {LabelExamples};

import TextField from "../src/text-field/examples.data";
export { TextField };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
