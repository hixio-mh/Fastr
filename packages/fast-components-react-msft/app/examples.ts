import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import DividerExample from "../src/divider/examples.data";
export { DividerExample };

import FlipperExample from "../src/flipper/examples.data";
export { FlipperExample };

import HeadingExample from "../src/heading/examples.data";
export { HeadingExample };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };
