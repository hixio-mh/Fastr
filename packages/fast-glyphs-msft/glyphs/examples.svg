<svg width="16" height="16" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="Canvas">
<g id="Group">
<g id="Vector">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.3 5.8L 2.3 3.9L 0.6 4.8L -3.8147e-07 3.8L 1.7 2.9L 0.1 2L 0.7 1L 2.2 1.9L 2.2 2.38419e-08L 3.3 2.38419e-08L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(10.3999 1.09998)"/>
</g>
<g id="Vector_2">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.2 5.8L 2.2 3.9L 0.6 4.8L 4.76837e-08 3.8L 1.6 2.9L 4.76837e-08 2L 0.6 1L 2.2 1.9L 2.2 -1.90735e-07L 3.3 -1.90735e-07L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(1.19995 10.2)"/>
</g>
<g id="Vector_3">
<path d="M 0 8L 0 0L 8 0L 8 8L 0 8ZM 1 1L 1 7L 7 7L 7 1L 1 1ZM 14.4 13.1L 16 14L 15.4 15L 13.8 14.1L 13.8 16L 12.7 16L 12.7 14.1L 11 15.1L 10.4 14.1L 12 13.2L 10.4 12.3L 11 11.3L 12.6 12.2L 12.6 10.3L 13.7 10.3L 13.7 12.2L 15.3 11.3L 15.9 12.3L 14.4 13.1Z"/>
</g>
</g>
</g>
</svg>