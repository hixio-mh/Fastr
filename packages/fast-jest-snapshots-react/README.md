# FAST Jest snapshots React
`@microsoft/fast-jest-snapshots-react` is a small utility library for creating snapshot Jest tests for React components.
Given a single component, the `generateSnapshots` function will iterate over a set of prop data and create a snapshot test for each `prop` instance.

## Usage
To use, import the `ISnapshotTestSuite` interface and the `generateSnapshots` function from `@microsoft/fast-jest-snapshots-react`:

```ts
import {generateSnapshots, ISnapshotTestSuite} from "@microsoft/fast-jest-snapshots-react";
```

Then, create you test data snapshot suite, where every index of `data` is a valid prop object for your `component`:
```ts
// in a Jest test file
import {MyReactComponent, IMyReactComponentProps} from "my-react-component.tsx";
import {ISnapshotTestSuite, generateSnapshots} from "@microsoft/fast-jest-snapshots-react";

const testSuite: ISnapshotTestSuite<IMyReactComponentProps> = {
    name: 'test component',
    component: MyCustomReactComponentConstructor,
    props: [
        {
            id: "test-id",
            text: "hello world",
            // other component props
        }
    ]
};

// Generate snapshots using the prop and component data you defined
generateSnapshots(testSuite);
```
