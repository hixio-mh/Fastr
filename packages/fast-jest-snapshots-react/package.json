{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "sideEffects": false,
  "version": "1.8.0",
  "author": {
    "name": "Microsoft",
    "url": "https://discord.gg/FcSNfg4"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.8.0",
    "@types/jest": "^22.2.2",
    "@types/react": "^16.3.6",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
