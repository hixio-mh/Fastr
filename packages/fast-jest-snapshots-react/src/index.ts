import * as React from "react";
import * as renderer from "react-test-renderer";

/**
 * An interface describing component example objects used for snapshots and component testing.
 */
export interface ISnapshotTestSuite<T> {
    /**
     * The name of the component
     */
    name: string;

    /**
     * The JSON schema for the components data
     */
    schema: any;

    /**
     * The component constructor
     */
    component: React.ComponentClass<T>;

    /**
     * The detail view component data
     */
    detailData: T;

    /**
     * An array of prop instances for the component
     */
    data: T[];

    /**
     * Documentation for the component
     */
    documentation?: JSX.Element;
}

/**
 * Executes a single snapshot test given a component, component data, and a test title
 */
export function renderSnapshot<T>(data: T, component: React.ComponentClass<T>, title: string): void {
    test(title, (): void => {
        const renderedComponent: any = renderer.create(React.createElement(component, data));
        const componentJson: JSON = renderedComponent.toJSON();

        expect(componentJson).toMatchSnapshot();
    });
}

/**
 * Generate a set of snapshot tests given a snapshot suite
 */
export function generateSnapshots<T>(examples: ISnapshotTestSuite<T>): void {
    const data: T[] = examples.data;
    const component: React.ComponentClass<T> = examples.component;

    if (Array.isArray(data)) {
        data.forEach((example: T, index: number): void => {
            renderSnapshot(example, component, `${examples.name}: ${index}`);
        });
    }
}
