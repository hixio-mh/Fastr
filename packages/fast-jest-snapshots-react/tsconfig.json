{
    "compilerOptions": {
        "moduleResolution": "node",
        "module": "commonjs",
        "target": "ES6",
        "baseUrl": ".",
        "declaration": true,
        "jsx": "react",
        "outDir": "./dist"
    },
    "include": [
        "src/**/*"
    ]
}
