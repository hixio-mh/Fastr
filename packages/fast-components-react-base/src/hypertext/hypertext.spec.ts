import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";

describe("hypertext", (): void => {
    generateSnapshots(examples);
});
