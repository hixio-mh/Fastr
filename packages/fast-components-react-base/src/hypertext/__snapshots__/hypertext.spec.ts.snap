// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hypertext Hypertext: 0 1`] = `
<a
  className="hypertext"
  href="https://msdn.microsoft.com/en-us/"
>
  MSDN
</a>
`;
